src/watertap_contrib/seto/unit_models/surrogate/tests/test_vagmd_batch_surrogate.py

```diff
@@ -0,0 +1,484 @@
+import pytest
+from pyomo.environ import (
+    ConcreteModel,
+    Set,
+    value,
+    assert_optimal_termination,
+    units as pyunits,
+)
+import re
+from pyomo.network import Port
+from idaes.core import FlowsheetBlock, UnitModelCostingBlock
+from watertap_contrib.seto.unit_models.surrogate import VAGMDbatch_surrogate
+
+from watertap.property_models.seawater_prop_pack import SeawaterParameterBlock
+from watertap_contrib.seto.costing import SETOWaterTAPCosting
+from idaes.core.util.testing import initialization_tester
+from watertap.core.util.initialization import assert_no_degrees_of_freedom
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+    unused_variables_set,
+)
+from idaes.core.util.testing import initialization_tester
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    constraint_scaling_transform,
+    unscaled_variables_generator,
+    unscaled_constraints_generator,
+    badly_scaled_var_generator,
+)
+
+import idaes.logger as idaeslog
+import numpy as np
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+
+
+class TestVAGMD:
+    @pytest.fixture(scope="class")
+    def VAGMD_frame(self):
+
+        module_type = "AS7C1.5L"  # Module type
+        recovery_ratio = 0.5  # Target recovery rate
+        feed_salinity = 35  # Feed salinity (g/L)
+        temp_evap_in = 80  # Evaporator inelt temperature (degC)
+        feed_flow_rate = 600  # Feed flow rate (L/h)
+        temp_cond_in = 25  # Condensor inlet temperature (degC)
+        temp_tank = 25  # Initial temperature of the feed (degC)
+        batch_volume = 50  # Initial batch volume
+
+        # Calculate final salinity temporarily
+        final_salinity = feed_salinity / (1 - recovery_ratio)
+
+        # Estimate the initial membrane performance
+        (
+            permeate_flux_init,
+            temp_cond_out_init,
+            temp_evap_out_init,
+            Area,
+        ) = self._get_membrane_performance(
+            temp_evap_in,
+            feed_flow_rate,
+            temp_cond_in,
+            feed_salinity,
+            temp_tank,
+            module_type,
+            final_salinity,
+        )
+        permeate_flow_rate_init = (
+            permeate_flux_init * Area
+        )  # Initial permeate flow rate (L/h)
```

Comment:
If you put the comment above this line rather than at the end of it, black won't reformat it this way.